timescaledb/tutorials/analyze-bitcoin-blockchain/analyze-blockchain.md

```diff
@@ -0,0 +1,393 @@
+block to include more transactions.
+
+In the previous charts, you saw how mining fees are correlated to block
+weights, transaction volumes, etc. But now, let’s analyze the data from a
+different perspective. Miner revenue is not only made up of miner fees, it
+also includes block rewards after mining a new block. This reward is currently
+6.25 BTC and it gets halved every four years. Let’s analyze miner revenue!
+
+### What percentage of the average miner's revenue comes from fees vs. block rewards?
+Miners are incentivized to keep the network up and running because they earn
+fees and rewards after mining each block. But let’s look at how much of their
+revenue comes from these two sources.
+
+```sql
+WITH coinbase AS (
+   SELECT block_id, output_total AS coinbase_tx FROM transactions
+   WHERE is_coinbase IS TRUE and time > NOW() - INTERVAL '5 days'
+)
+SELECT
+   bucket as "time",
+   avg(block_fee_sat)*0.00000001 AS "fees",
+   FIRST((c.coinbase_tx - block_fee_sat), bucket)*0.00000001 AS "reward"
+FROM one_hour_blocks b
+INNER JOIN coinbase c ON c.block_id = b.block_id
+GROUP BY bucket
+ORDER BY 1;
+```
+![fees and block reward](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/revenue_ratio.png)
+(THIS CHART NEEDS AN UPDATE)
+
+This chart analyzes the last 5 days of average miner revenue. The left
+axis is a percentage value that indicates the portion of total revenue that
+comes from transaction fees (green) and block rewards (yellow). It might be
+surprising for some, but most miner revenue comes from block rewards
+(6.25BTC at the moment). The fees portion never went above 3 % in the
+last 5 days. 
+
+This kind of analysis can start discussions around the long-term fading of
+block rewards and how on-chain fees will need to go up in the future to
+incentivize the miners and sustain the network. (Note that the left axis is
+logarithmic-scale, so it’s easier to see the green "fees" portion.)
+
+### How does block weight affect miner fees?
+You’ve already seen that more transactions in a block mean it’s more expensive
+to mine. Is it the same with block weights? The more transactions a block has,
+the larger its size (or weight), so the block weight and mining fee should
+be tightly correlated. 
+
+This query uses a 12-hour moving average to calculate the block weight and
+block mining fee over time.
+
+```sql
+WITH stats AS (
+   SELECT
+       bucket,
+       stats_agg(block_weight, block_fee_sat) AS block_stats
+   FROM one_hour_blocks
+   WHERE bucket > NOW() - INTERVAL '5 days'
+   GROUP BY bucket
+)
+SELECT
+   bucket as "time",
+   average_y(rolling(block_stats) OVER (ORDER BY bucket RANGE '12 hours' PRECEDING)) AS "block weight",
+   average_x(rolling(block_stats) OVER (ORDER BY bucket RANGE '12 hours' PRECEDING))*0.00000001 AS "mining fee"
+FROM stats
+ORDER BY 1
+```
+
+![block weight and fees](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/weight_fees.png)
```

Comment:
```suggestion
![block weight and fees](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/weight_fees.png)
```
Am I misinterpreting something, or is this not the graph produced from the query? Graph is titled 4-week moving average, and x-axis spans less than a day

Also this section seems very similar to the weight vs. fee section above, but using a rolling average. Is there a reason for having both, or should they be consolidated into one?

Reply:
correct good catch let me create the proper chart